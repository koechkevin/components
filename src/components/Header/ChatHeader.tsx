import { IconDefinition } from '@fortawesome/pro-solid-svg-icons';
import { Row } from 'antd';
import React, { FC } from 'react';

import { Icon } from '../Icon';
import Label from '../Label/Label';
import { Text } from '../Typography';
import styles from './ChatHeader.module.scss';

interface Props {
  statusIcon?: IconDefinition;
  iconColor?: string;
  jobPosition?: string;
  chatRoomStatus?: string;
}

const ChatHeader: FC<Props> = (props) => {
  const { statusIcon, iconColor, jobPosition, chatRoomStatus } = props;

  return (
    <Row align="middle" type="flex" className={styles.chatHeaderWrapper}>
      {statusIcon && <Icon className={styles.chatStatusIcon} icon={statusIcon} color={iconColor} />}
      {jobPosition && (
        <Text ellipsis style={{ paddingRight: 8 }} className={styles.jobPosition}>
          {jobPosition}
        </Text>
      )}
      {chatRoomStatus && <Label>{chatRoomStatus}</Label>}
    </Row>
  );
};

export default ChatHeader;
