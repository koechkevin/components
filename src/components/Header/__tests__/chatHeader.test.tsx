import { faCircle } from '@fortawesome/pro-light-svg-icons';
import { mount } from 'enzyme';
import React from 'react';

import ChatHeader from '../ChatHeader';

describe('Tests for the ChatHeader component', () => {
  const props = {
    statusIcon: faCircle,
    iconColor: 'red',
    jobPosition: 'CEO',
    chatRoomStatus: 'open',
  };

  const wrapper = mount(<ChatHeader {...props} />);

  it('should render ChatHeader without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the props', () => {
    expect(wrapper.props().iconColor).toEqual('red');
    expect(wrapper.props().jobPosition).toEqual('CEO');
    expect(wrapper.props().chatRoomStatus).toEqual('open');
  });
});
