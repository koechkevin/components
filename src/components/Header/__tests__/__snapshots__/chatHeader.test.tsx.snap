// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the ChatHeader component should render ChatHeader without crashing 1`] = `ReactWrapper {}`;
